ancestrymanager/ancestryutil.go

```diff
@@ -104,6 +112,17 @@ func getFolderFromResource(tfData resources.TerraformResourceData) (string, bool
 	if ok {
 		return folderID.(string), ok
 	}
+	// folder name is store in display_name for google_folder
+	folderID, ok = tfData.GetOk("display_name")
```

Comment:
There are a number of "display_name" fields on different resources that generally hold "human-friendly" names for resources, so we can't rely on it to hold a folder identifier. Even for folder - I am pretty sure the display_name isn't used in the ancestry. The value returned from this function will be used to cache the ancestry of the folder, so we need to be able to rely on it to be in the standard format - which is `folders/XXXXX`.

Looking at the resource - you'll probably need to either:

1. use `folder.id` and check with a regex that it exactly matches the expected format (because every terraform resource has an `id` field
2. use `folder_id` but add the `folders/` prefix back in - because the folder resource [strips it out](https://github.com/hashicorp/terraform-provider-google/blob/c0c2e42de049c5a0c37ba460626ac7ae113c1f0e/google/resource_google_folder.go#L138).

No matter which you choose, you'll need to make sure (with new test cases) that folder will return a valid ancestry even if folder_id is present on the resource with just a number in it. (Currently, this function assumes that `folder_id` contains something like `folders/XXXX` because on most resources, that's correct.)

Reply:
I have removed the code fetching {display_name} as an ancestor for google_folder. As discussed in offline conversation, ancestry path will be determined by {parent} field when the {folder_id} for google_folder is not yet set.